src/components/Button/Button.tsx

```diff
@@ -299,9 +303,10 @@ export function Button({
     <UnstyledButton {...commonProps} {...linkProps} {...actionProps}>
       <span className={styles.Content}>
         {spinnerSVGMarkup}
-        {iconMarkup}
+        {iconPlacement === 'before' && iconMarkup}
         {childMarkup}
-        {disclosureMarkup}
+        {iconPlacement === 'after' && iconMarkup}
+        {!connectedDisclosureMarkup && disclosureMarkup}
```

Comment:
Lots of short circuiting here i.e. `&&`. It would be nice to have this stuff a bit more readable